packages/composable-controller/src/ComposableController.ts

```diff
@@ -113,15 +154,18 @@ export class ComposableController extends BaseController<
           [name]: childState,
         }));
       });
-    } else {
-      this.messagingSystem.subscribe(
-        `${String(name)}:stateChange`,
-        (childState: Record<string, unknown>) => {
+    } else if (isBaseController(controller)) {
+      this.messagingSystem.subscribe(`${name}:stateChange`, (childState) => {
+        if (isValidJson(childState)) {
```

Comment:
Since the child controller is validated as BaseControllerV2, its state type is guaranteed to be `Record<string, Json>`. It's therefore safe to narrow `childState` from `Record<string, unknown>`.